core/src/test/java/com/expediagroup/streamplatform/streamregistry/core/services/ProcessServiceTest.java

```diff
@@ -201,4 +218,22 @@ public void delete_multi() {
     inOrder.verify(processBindingService).delete(binding2);
     inOrder.verify(processRepository).delete(entity);
   }
+
+  private Process createTestProcess() {
+    Process p = new Process();
+    List<ProcessInputStream> inputs = new ArrayList<>();
+    ProcessInputStream pis = new ProcessInputStream(new StreamKey("inputDomain","streamInputName",1), new ObjectMapper().createObjectNode());
+    inputs.add(pis);
+    p.setInputs(inputs);
+    List<ProcessOutputStream> outputs = new ArrayList<>();
+    ProcessOutputStream pos = new ProcessOutputStream(new StreamKey("outputDomain","streamOutputName",1), new ObjectMapper().createObjectNode());
+    outputs.add(pos);
+    p.setOutputs(outputs);
+    p.setKey(new ProcessKey("domain","name"));
+    List<ZoneKey> zones = new ArrayList<>();
+    zones.add(new ZoneKey("aws_us_east_1"));
+    p.setZones(zones);
+    return p;
+  }
+
 }
```

Comment:
There needs to be test cases in here that will ensure that a process.save fails if a consumer or producer authorize check fails. You can simulate that with a mock of the consumer/producer service and just fail the test authorize call. Although I am not sure what type of exception is thrown, so you will have to look that up.

Reply:
updated